scipy/ndimage/tests/test_filters.py

```diff
@@ -2986,7 +2990,19 @@ def test_edge_cases(self, xp):
         ref = ndimage.vectorized_filter(input, function, size=21)
         xp_assert_close(res, ref)
 
-    def test_gh23046(self, xp):
+    def test_gh23046_feature(self, xp):
+        # The intent of gh-23046 was to always allow `size` to be a scalar.
+        rng = np.random.default_rng(45982734597824)
+        img = xp.asarray(rng.random((5, 5)))
+
+        ref = ndimage.vectorized_filter(img, xp.mean, size=2)
+        res = ndimage.vectorized_filter(img, xp.mean, size=2, axes=(0, 1))
+        xp_assert_close(res, ref)
+
+        ref = ndimage.vectorized_filter(img, xp.mean, size=(2,), axes=(0,))
+        res = ndimage.vectorized_filter(img, xp.mean, size=2, axes=0)
```

Comment:
@mdhaber was the `xp_assert_close` left out on purpose here? At first I thought it might be just make sure things don't error in these statements, but then assigning to `ref` and `res` may not be of much use?

Reply:
Oops, no : / They should be compared. I can submit a PR tonight.

Reply:
I already have it on the release branch locally, so no rush, maybe just do it on `main` when you have time so I don't have to forward port it